### PR TITLE
Added Create an IBFT 2.0 network tutorial

### PR DESCRIPTION
**Pull request by MadelineMurray**
Added Create an IBFT 2.0 network tutorial

This needs to be updated when the RLP encode command has been completed but the steps should all be there.

### CODE REVIEW

File: docs/Tutorials/Create-IBFT-Network.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+```bash tab="Windows"
+pantheon --data-path=Node-1-data-path public-key export-address --to=Node-1-data-path\nodeAddress1
+```
+
+### 4. RLP Encode Extra Data 
+
+The `extraData` property in IBFT 2.0 genesis files is an RLP encoding of `[32 Bytes Vanity, List<Validators>, No Votes, Round=Int(0), 0 Seals]`. 
+
+Use the Pantheon subcommand `encode` to generate the `extraData`.
```
Comment (NicolasMassart) [inline docs/Tutorials/Create-IBFT-Network.md]:
```suggestion
Use the Pantheon [`rlp encode`](../Reference/Pantheon-CLI-Syntax/#rlp) subcommand to generate the `extraData` RLP string to add in the genesis file.
```

File: docs/Tutorials/Create-IBFT-Network.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+```
+
+In `extraData`, include the [RLP encoded data that includes the initial validator](#4-rlp-encode-extra-data). 
+
+!!! example
+    
+    ```json
+    {
+      ...
+    "extraData":"UPDATE",
```
Comment (NicolasMassart) [inline docs/Tutorials/Create-IBFT-Network.md]: If you use the 3 addresses already in the genesis it your give an extradata rlp string like : 
```suggestion
    "extraData":"0xf869a00000000000000000000000000000000000000000000000000000000000000000f83f94fe3b557e8fb62b89f4916b721be55ceb828dbd7394627306090abab3a6e1400e9345bc60c78a8bef5794f17f52151ebef6c7334fad080c5704d77216b732808400000000c0"
```

From the Json : 
```
[
"fe3b557e8fb62b89f4916b721be55ceb828dbd73",
"627306090abaB3A6e1400e9345bC60c78a8BEf57",
"f17f52151EbEF6C7334FAD080c5704D77216b732"
]
```

File: docs/Tutorials/Create-IBFT-Network.md [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+
+!!! info 
+    Pantheon does not implement [private key management](../Using-Pantheon/Account-Management.md).
+
+## Stop Nodes
+
+When finished using the private network, stop all nodes using ++ctrl+c++ in each terminal window. 
+
+!!!tip
+    To restart the IBFT 2.0 network in the future, start from [7. Start First Node as Bootnode](#7-start-first-node-as-bootnode).
```
Comment (macfarla) [inline docs/Tutorials/Create-IBFT-Network.md]: s/7/8

File: docs/Tutorials/Create-IBFT-Network.md [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+
+## Next Steps 
+
+Look at the logs displayed to confirm Node-1 is producing blocks and Node-2 and Node-3 are importing blocks. 
+
+Use the [IBFT API to add](../Consensus-Protocols/Clique.md#adding-and-removing-signers) Node-2 or Node-3 as a validator. 
+
+!!! note
+    To add Node-2 or Node-3 as a validator you need the [node address as when specifying Node-1](#2-get-address-for-node-1) as the initial validator. 
+
```
Comment (macfarla) [inline docs/Tutorials/Create-IBFT-Network.md]: I got a bit lost in the adding of validators but that isn't part of this PR

File: docs/Tutorials/Create-IBFT-Network.md (hunk 1 above)
Comment (MadelineMurray) [reply docs/Tutorials/Create-IBFT-Network.md]: Fixed

File: docs/Tutorials/Create-IBFT-Network.md (hunk 2 above)
Comment (MadelineMurray) [reply docs/Tutorials/Create-IBFT-Network.md]: Fixed

File: docs/Reference/Pantheon-CLI-Syntax.md [hunk 5]
```diff
@@ -1014,15 +1014,15 @@ $ pantheon public-key export [--to=<key-file>]
...
 ```
 
 ```bash tab="Example (to file)"
-$ pantheon public-key export --to=/home/me/me_project/not_precious_pub_key
+$ pantheon --data-path=<node data path> public-key export --to=/home/me/me_project/not_precious_pub_key
 ```
 
-Outputs the node public key to standard output or write it in the specified file if option 
-`--to=<key-file>` is defined. 
+Outputs the node public key to standard output or writes it in the specified file if
```
[the remark this answers is not in the document]
Comment (MadelineMurray) [reply docs/Reference/Pantheon-CLI-Syntax.md]: Fixed

File: docs/Reference/Pantheon-CLI-Syntax.md [hunk 6]
```diff
@@ -1031,15 +1031,15 @@ $ pantheon public-key export-address [--to=<address-file>]
...
 ```
 
 ```bash tab="Example (to file)"
-$ pantheon public-key export-address --to=/home/me/me_project/me_node_address
+$ pantheon --data-path=<node data path> public-key export-address --to=/home/me/me_project/me_node_address
 ```
 
-Outputs the node public key address to standard output or write it in the specified file if option 
-`--to=<key-file>` is defined. 
+Outputs the node public key address to standard output or writes it in the specified file if
```
[the remark this answers is not in the document]
Comment (MadelineMurray) [reply docs/Reference/Pantheon-CLI-Syntax.md]: Fixed

File: docs/Tutorials/Create-IBFT-Network.md [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+        ├── key
+        ├── publicKeyNode1
+```
+      
+The `database` directory contains the blockchain data. 
+
+### 3. Get Node Addresses 
+
+In IBFT 2.0 networks, the address of at least one initial validator must be included in the genesis file in the
+RPL encoded `extraData` string. For this network, we will use Node-1 as the initial validator. This requires obtaining the address for Node-1.
```
[the remark this answers is not in the document]
Comment (MadelineMurray) [reply docs/Tutorials/Create-IBFT-Network.md]: Fixed

File: docs/Tutorials/Create-IBFT-Network.md [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+
+```bash tab="MacOS"
+pantheon --data-path=Node-1-data-path public-key export-address --to=Node-1-data-path/nodeAddress1
+```
+
+```bash tab="Windows"
+pantheon --data-path=Node-1-data-path public-key export-address --to=Node-1-data-path\nodeAddress1
+```
+
+To vote new validators in once the network is running, the node address is required. In the `Node-2`
```
[the remark this answers is not in the document]
Comment (MadelineMurray) [reply docs/Tutorials/Create-IBFT-Network.md]: Reworded

File: docs/Tutorials/Create-IBFT-Network.md [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+
+In `extraData`, copy the [RLP encoded data](#5-rlp-encode-extra-data) from the `rlpEncodedExtraData` file. 
+
+!!! example
+    
+    ```json
+    {
+      ...
+    "extraData":0xf83ea00000000000000000000000000000000000000000000000000000000000000000d5949b9f91039843450927b0043ae71cd803e0db0c30808400000000c0,
+      ...
```
[the remark this answers is not in the document]
Comment (MadelineMurray) [reply docs/Tutorials/Create-IBFT-Network.md]: Added

File: docs/Tutorials/Create-IBFT-Network.md [hunk 10]
```diff
@@ -0,0 +1,309 @@
...
+  "id" : 1,
+  "result" : "0x2"
+}
+```
+
+## Next Steps 
+
+Look at the logs displayed to confirm Node-1 is producing blocks and Node-2 and Node-3 are importing blocks. 
+
+Use the [IBFT API to add](../Consensus-Protocols/Clique.md#adding-and-removing-signers) Node-2 or Node-3 as a validator.
```
[the remark this answers is not in the document]
Comment (MadelineMurray) [reply docs/Tutorials/Create-IBFT-Network.md]: Fixed

File: docs/Tutorials/Create-IBFT-Network.md (hunk 3 above)
Comment (MadelineMurray) [reply docs/Tutorials/Create-IBFT-Network.md]: Fixed

File: docs/Tutorials/Create-IBFT-Network.md (hunk 4 above)
Comment (MadelineMurray) [reply docs/Tutorials/Create-IBFT-Network.md]: I'll look at making this clearer after the release.

File: docs/Consensus-Protocols/IBFT.md [hunk 11]
```diff
@@ -39,7 +39,7 @@ Properties specific to IBFT 2.0 are:
 * `blockperiodseconds` - Minimum block time in seconds. 
 * `epochlength` - Number of blocks after which to reset all votes.
 * `requesttimeoutseconds` - Timeout for each consensus round before a round change. 
-* `extraData` - RLP([32 Bytes Vanity, List<Validators>, No Votes, Round=Int(0), 0 Seals])
+* `extraData` - `RLP([32 Bytes Vanity, List<Validators>, No Votes, Round=Int(0), 0 Seals])`
```
[the remark this answers is not in the document]
Comment (MadelineMurray) [reply docs/Consensus-Protocols/IBFT.md]: Fixed

File: docs/Reference/Pantheon-CLI-Syntax.md [hunk 12]
```diff
@@ -1014,15 +1014,15 @@ $ pantheon public-key export [--to=<key-file>]
...
 ```
 
 ```bash tab="Example (to file)"
-$ pantheon public-key export --to=/home/me/me_project/not_precious_pub_key
+$ pantheon --data-path=<node data path> public-key export --to=/home/me/me_project/not_precious_pub_key
 ```
 
-Outputs the node public key to standard output or write it in the specified file if option 
-`--to=<key-file>` is defined. 
+Outputs the node public key to standard output or writes it to the specified file if
```
[the remark this answers is not in the document]
Comment (MadelineMurray) [reply docs/Reference/Pantheon-CLI-Syntax.md]: I don't know - might be in the CLI guidelines @jakehaugen? @NicolasMassart?

File: docs/Reference/Pantheon-CLI-Syntax.md [hunk 13]
```diff
@@ -1031,15 +1031,15 @@ $ pantheon public-key export-address [--to=<address-file>]
 ```
 
 ```bash tab="Example (to standard output)"
-$ pantheon public-key export-address
+$ pantheon --data-path=<node data path> public-key export-address
```
[the remark this answers is not in the document]
Comment (MadelineMurray) [reply docs/Reference/Pantheon-CLI-Syntax.md]: Again - unsure. @jakehaugen @NicolasMassart?

File: docs/Tutorials/Create-IBFT-Network.md [hunk 14]
```diff
@@ -0,0 +1,309 @@
...
+
+### 1. Create Folders 
+
+Each node requires a data directory for the blockchain data. When the node is started, the [node key](../Configuring-Pantheon/Node-Keys.md) is saved in this directory. 
+
+Create directories for your private network, each of the three nodes, and a data directory for each node: 
+
+```bash
+IBFT-Network/
+├── Node-1
```
[the remark this answers is not in the document]
Comment (MadelineMurray) [reply docs/Tutorials/Create-IBFT-Network.md]: FIxed

File: docs/Tutorials/Create-IBFT-Network.md [hunk 15]
```diff
@@ -0,0 +1,309 @@
...
+│   ├── Node-1-data-path
+├── Node-2
+│   ├── Node-2-data-path
+└── Node-3
+    ├── Node-3-data-path
+```
+
+### 2. Get Public Key for Node-1
```
[the remark this answers is not in the document]
Comment (MadelineMurray) [reply docs/Tutorials/Create-IBFT-Network.md]: Fixed

File: docs/Tutorials/Create-IBFT-Network.md [hunk 16]
```diff
@@ -0,0 +1,309 @@
...
+
+In the `Node-1` directory, use the [`public-key export`](../Reference/Pantheon-CLI-Syntax.md#public-key) subcommand to write 
+the [node public key](../Configuring-Pantheon/Node-Keys.md) to the specified file (`publicKeyNode1` in this example):
+
+```bash tab="MacOS"
+pantheon --data-path=Node-1-data-path public-key export --to=Node-1-data-path/publicKeyNode1
+```
+
+```bash tab="Windows"
+pantheon --data-path=Node-1-data-path public-key export --to=Node-1-data-path\publicKeyNode1
```
[the remark this answers is not in the document]
Comment (MadelineMurray) [reply docs/Tutorials/Create-IBFT-Network.md]: Fixed

**Pull request merged**